tests/appsec/test_automated_user_and_session_tracking.py

```diff
@@ -0,0 +1,186 @@
+# Unless explicitly stated otherwise all files in this repository are licensed under the the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2024 Datadog, Inc.
+
+from utils import context
+from utils import features
+from utils import interfaces
+from utils import remote_config as rc
+from utils import rfc
+from utils import scenarios
+from utils import weblog
+
+
+def login_data(context):
+    """In Rails the parametesr are group by scope. In the case of the test the scope is user.
+   The syntax to group parameters in a POST request is scope[parameter]
+   """
+    username_key = "user[username]" if "rails" in context.weblog_variant else "username"
+    password_key = "user[password]" if "rails" in context.weblog_variant else "password"
+    return {username_key: "test", password_key: "1234"}
+
+
+@rfc("https://docs.google.com/document/d/1Z0kQ5f_GZR5thtwQjC6C_iY9C85Q-Cf-vFj9h_xd0zE")
+@features.user_monitoring
+class Test_Automated_User_Tracking:
+    def setup_user_tracking(self):
+        self.r_login = weblog.post("login?auth=local", data=login_data(context))
+        self.cookies = self.r_login.cookies
+        self.r_user = weblog.get("user/me", cookies=self.cookies,)
+
+    def test_user_tracking(self):
+        assert self.r_login.status_code == 200
+        assert self.r_user.status_code == 200
+
+        for _, trace, span in interfaces.library.get_spans(request=self.r_user):
+            meta = span.get("meta", {})
+            assert meta["usr.id"] == "social-security-id"
```

Comment:
>_dd.appsec.usr.id: this internal tag must be added only when the user ID was collected automatically (ie: not collected through the SDK). This ensures that accuracy analysis can be performed, taking advantage of users who are already collecting this information themselves.

```suggestion
            assert meta["usr.id"] == "social-security-id"
            assert meta["_dd.appsec.usr.id"] == "social-security-id"
```

Reply:
This is from the new RFC not yet approved. The behavior on the tests matches the last [one](https://docs.google.com/document/d/1Z0kQ5f_GZR5thtwQjC6C_iY9C85Q-Cf-vFj9h_xd0zE/edit?tab=t.0#heading=h.e9h83guhrzuo) which is released already in python and java. I had in mind to revamp all the tests once the new RFC it's approved.

Reply:
The thing is if we don't add it now, if we want to add it later, we'll have to create a new test, otherwise we'll lose the coverage for the previous tests, so in the end it will be the same, but if we do it later, we run the risk of forgetting.
But no super strong opinion

Reply:
As discussed, I'll keep the PR open as a draft until the RFC gets approved, then I'll add all the needed assertions, thanks for the feedback :wink: